### PR TITLE
Remove QuorumReplicationProtocol

### PR DESCRIPTION
**Pull request by Maithem**
## Overview
Most of the code is non functional and its becoming costly to
manage it while adding new code.

Why should this be merged: Improves code management 

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

**Review by xcchang — APPROVED** [review]
LGTM, will rebase the Protobuf LogUnit accordingly.

File: runtime/src/main/java/org/corfudb/runtime/clients/LogUnitHandler.java [hunk 1]
```diff
@@ -16,12 +13,10 @@
 import org.corfudb.protocols.wireprotocol.ReadResponse;
 import org.corfudb.protocols.wireprotocol.TailsResponse;
 import org.corfudb.runtime.exceptions.DataCorruptionException;
-import org.corfudb.runtime.exceptions.DataOutrankedException;
 import org.corfudb.runtime.exceptions.OutOfSpaceException;
 import org.corfudb.runtime.exceptions.OverwriteCause;
 import org.corfudb.runtime.exceptions.OverwriteException;
 import org.corfudb.runtime.exceptions.TrimmedException;
-import org.corfudb.runtime.exceptions.ValueAdoptedException;
```
Comment (xcchang) [inline runtime/src/main/java/org/corfudb/runtime/clients/LogUnitHandler.java:20]: Looks like we can also delete `ValueAdoptedException` file?

Comment (xcchang) [reply runtime/src/main/java/org/corfudb/runtime/clients/LogUnitHandler.java:20]: Or it could be handled in LogUnit RPC PR

Comment (Maithem) [reply runtime/src/main/java/org/corfudb/runtime/clients/LogUnitHandler.java:20]: I removed it.

Comment (Maithem) [reply runtime/src/main/java/org/corfudb/runtime/clients/LogUnitHandler.java:20]: Also it seems like you can remove `OutOfSpaceException` too its not being used, so it shouldn't be in the protobuf definitions.

Comment (zfrenette) [reply runtime/src/main/java/org/corfudb/runtime/clients/LogUnitHandler.java:20]: It hasn't been included in the Protobuf definitions.

**Review by zhangn49 — APPROVED** [review]

**Pull request merged**